app/javascript/components/widgets/numbers/falling-blocks.jsx

```diff
@@ -43,10 +46,20 @@ class Scene extends React.Component {
       const loadedWorld = this.res.resurrect(world);
       Engine.merge(this.engine, { world: loadedWorld });
     }
+    Composite.allBodies(this.engine.world)
+      .filter(body => body.isStatic && body.render.sprite.texture)
+      .map(b => this.addToOverlay(b));
+    Events.on(this.engine, 'beforeUpdate', b => this.prepareOverlay(b));
     this.addBlocks();
   }
 
   componentWillUnmount() {
+    const allBodies = Composite.allBodies(this.engine.world);
+    for (let i = 0; i < allBodies.length; i += 1) {
+      if (allBodies[i].onOverlay) {
+        allBodies[i].onOverlay = false;
```

Comment:
This is a strange name for a boolean. `on<something>` is generally a callback. `is<something>` / `has<something>` are better boolean indicators